fea-rs/src/compile/validate.rs

```diff
@@ -1283,20 +1286,27 @@ impl<'a> ValidationCtx<'a> {
                     self.error(item.axis_tag().range(), "unknown axis");
                     continue;
                 };
-                let val = item.value().parse_signed() as i32;
-                let min = axis_info.min_value.to_i32();
-                let max = axis_info.max_value.to_i32();
-                if val < min {
-                    self.error(
-                        item.value().range(),
-                        format!("value below minimum allowed for axis ({min})"),
-                    );
-                }
-                if val > max {
-                    self.error(
-                        item.value().range(),
-                        format!("value greater than maximum allowed for axis ({max})"),
-                    );
+                let val = item.value().parse();
+                match val {
+                    super::AxisLocation::User(val) => {
+                        let min = axis_info.min_value;
+                        let max = axis_info.max_value;
+                        if val < min || val > max {
+                            self.error(
+                                item.value().range(),
+                                format!("value exceeds expected range ({min}, {max})"),
+                            );
+                        }
+                    }
+                    super::AxisLocation::Design(_) => (), // we don't have info to validate this
```

Comment:
> we don't have info to validate this

if we care about design locations, maybe the VariationInfo trait can have an additional method similar to the current `normalize_coordinate`, that instead maps from user to design coordinates, then one can use the min/max (user) values from axis_info and convert them to design coordinates and validate locations are in range

Reply:
My feeling on this stuff right now is that we're currently iterating and trying to figure out what patterns make sense, and then we can nail down a good API at some point in the future. That's a reasonable suggestion, though!